source/mesh_refinement/isotherms.cc

```diff
@@ -0,0 +1,343 @@
+            std::vector<std::string> isotherms_outer_loop = Utilities::split_string_list(prm.get ("List of isotherms"),';');
+            // process the List of isotherms to get all data out of the structure
+            number_of_isotherms = isotherms_outer_loop.size();
+            isotherms.resize(number_of_isotherms,std::vector<std::string>(4,"0"));
+            // loop through all given isotherms
+            for ( unsigned int it1 = 0; it1 != number_of_isotherms; ++it1 )
+            {
+                std::vector<std::string> isotherms_inner_loop = Utilities::split_string_list(isotherms_outer_loop[it1]);
+                // The minimum and maximum refinemnt levels respecively
+                isotherms[it1][0]=isotherms_inner_loop[0];
+                isotherms[it1][1]=isotherms_inner_loop[1];
+                // The temperatures for the min and max isotherms respectively
+                isotherms[it1][2]=isotherms_inner_loop[2];
+                isotherms[it1][3]=isotherms_inner_loop[3];
+
+            }
+        }
+        prm.leave_subsection();
+    }
+    prm.leave_subsection();
+}
+}
+}
+
+// explicit instantiations
+namespace aspect
+{
+namespace MeshRefinement
+{
+ASPECT_REGISTER_MESH_REFINEMENT_CRITERION(Isotherms,
+        "isotherms",
+        "A mesh refinement criterion that ensures a "
+        "maximum and minimum refinement level between "
+        "two temperatures (isotherms), with the posibility "
+        "to exclude a composition from this criterion. "
+        "To accomplish this there are two parameters "
+        "available: 'Exclude composition' and 'List of "
+        "isotherms'. The first parameter takes the "
+        "number of the compositional field and excludes "
+        "it from the min/max refinment level prodedure. "
+        "The second parameters takes a list of isoterm "
+        "parameters for one isoterm separated by a ';'. "
+        "Each line of isoterm parameters contains four "
+        "subparameters. The first subparameter sets "
+        "the minimun refinement level for this isoterm, "
+        "the second subparameter sets the maximum "
+        "refiment level for this isoterm, the third "
+        "subparameter sets the minimum temperature for "
+        "this isoterm and the fourth subparameter sets "
+        "the maximum termperature for the isoterm. "
+        "The minimum and maximum level can be indicated "
+        "by the absolute number of the refinement level "
+        "or by using the key words 'min' or 'max', "
+        "corrosponding respecively to the set minimum "
+        "and maximum refinment level. The keywords "
+        "'min' and 'max' may also be used in combination "
+        "with addition or substraction, e.g. 'max-1' "
+        "or 'min+1'. For formating it is usefull to "
+        "make use of the build in option of ASPECT "
+        "to split parameters over several line by "
+        "putting ' \\' at the end of the line. "
```

Comment:
Things here will be escaped more than one: when going from source code to internal string representations, and again into latex. Just say "putting a backslash at the end of the line."